js/app.js

```diff
@@ -48,17 +48,24 @@ angular.module('PathOfDamage', [])
   };
 
   $scope.updateHitRow = function (index) {
-    if (index === $scope.hits.length - 1 || $scope.hits[index].hit !== null) {
+    if ($scope.hits[index].hit) {
       $scope.hits[index] = calcDamage($scope.hits[index].hit);
       if ($scope.hits[$scope.hits.length - 1].hit !== null) {
         $scope.hits.push({hit: null});
       }
-    } else {
+    } else if (!$scope.hits[index].focus && index !== $scope.hits.length - 1){
       $scope.hits.splice(index, 1);
     }
     serializeData();
   };
 
+  $scope.updateHitFocus = function (index, hasFocus) {
+    $scope.hits[index].focus = hasFocus;
+    if (!hasFocus && !$scope.hits[index].hit) {
+      $scope.updateHitRow(index);
```

Comment:
we should probably split the updateHit method up at this point, since we don't need to check for updating calculation values when losing or gaining focus, that's entirely based on typing. ng-change should only update calculations and possibly add a row, ng-blur should check if it needs to delete a row, then we don't need ng-focus, and we don't need to store focus on the hit either.